### PR TITLE
fix: prevented dashed props in svg path

### LINKED ISSUE #648: TypeScript should throw errors if people try to use standard svg properties on icons like `fill-rule`, `clip-rule`
I think these should be `fillRule`, `clipRule` (TS should ideally be warning these). Similarly at other places.

_Originally posted by @divyanshu013 in https://github.com/razorpay/blade/pull/645#discussion_r948940625_

### PR DESCRIPTION
**Pull request by anuraghazra**
Throws error if consumers add any dashed props in the JSX. 

image

fixes #648

### CODE REVIEW

File: packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIcon.tsx [hunk 1]
```diff
@@ -12,8 +12,8 @@ const CheckedIcon = ({ color }: { color: string }) => {
   return (
     <Svg width="16px" height="16px" viewBox="0 0 16 16" fill="none">
       <Path
-        fill-rule="evenodd"
-        clip-rule="evenodd"
```
Comment (anuraghazra) [inline packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIcon.tsx:16]: Yep. It works.

**Comment by anuraghazra** [comment]
OMMMMGG What happened here 🤯 

image

File: packages/blade/src/components/Icons/_Svg/Path/Path.d.ts [hunk 2]
```diff
@@ -1,3 +1,7 @@
+export type PreventDashedProps<T> = {
+  [K in keyof T]: K extends `${string}-${string}` ? never : T[K];
```
Comment (divyanshu013) [inline packages/blade/src/components/Icons/_Svg/Path/Path.d.ts:2]: :godmode:

**Review by divyanshu013 — COMMENTED** [review]
Check the diff in snapshot please, it seems too large with 921K lines

**Comment by anuraghazra** [comment]
> Check the diff in snapshot please, it seems too large with 921K lines

Yeah fixed lol. I forgot to do .toJSON()

File: packages/blade/src/components/Icons/_Svg/Path/Path.d.ts [hunk 3]
```diff
@@ -1,3 +1,7 @@
+export type PreventDashedProps<T> = {
```
Comment (divyanshu013) [inline packages/blade/src/components/Icons/_Svg/Path/Path.d.ts:1]: Optional: Maybe `NonDashedProps` can be better term for this

**Review by divyanshu013 — APPROVED** [review]
This PR will save lot of hours 🎖️

File: packages/blade/src/components/Icons/_Svg/Path/Path.d.ts (hunk 3 above)
Comment (anuraghazra) [reply packages/blade/src/components/Icons/_Svg/Path/Path.d.ts:1]: Better. 👍

**Pull request merged**